mhkit/power/characteristics.py

```diff
@@ -1,121 +1,185 @@
 
     return P
 
-def ac_power_three_phase(voltage, current, power_factor, line_to_line=False):
+def ac_power_three_phase(voltage, current, power_factor, line_to_line=False, to_pandas=True):
     """
     Calculates magnitude of active AC power from line to neutral voltage and current 
 
     Parameters
     -----------
-    voltage: pandas DataFrame
-        Time-series of three phase measured voltage [V] indexed by time
-    current: pandas DataFrame 
-        Time-series of three phase measured current [A] indexed by time
+    voltage: pandas Series, pandas DataFrame, xarray DataArray, or xarray Dataset
+        Measured DC voltage [V] indexed by time
+
+    current: pandas Series, pandas DataFrame, xarray DataArray, or xarray Dataset
+        Measured three phase current [A] indexed by time
+
     power_factor: float 
         Power factor for the efficiency of the system
-    line_to_line: bool
+
+    line_to_line: bool (Optional)
         Set to true if the given voltage measurements are line_to_line
-    
+
+    to_pandas: bool (Optional)
+        Flag to save output to pandas instead of xarray. Default = True.
+
     Returns
     --------
-    P: pandas DataFrame
+    P: pandas DataFrame or xarray Dataset
         Magnitude of active AC power [W] indexed by time with Power column 
     """
-    if not isinstance(voltage, pd.DataFrame):
-        raise TypeError(f'voltage must be of type pd.DataFrame. Got: {type(voltage)}')
-    if not isinstance(current, pd.DataFrame):
-        raise TypeError(f'current must be of type pd.DataFrame. Got: {type(current)}')
-    if not len(voltage.columns) == 3:
+    if not isinstance(voltage, (pd.Series, pd.DataFrame, xr.DataArray, xr.Dataset)):
+        raise TypeError('voltage must be of type pd.Series, pd.DataFrame, ' + 
+                        f'xr.DataArray, or xr.Dataset. Got{type(voltage)}')
+    if not isinstance(current, (pd.Series, pd.DataFrame, xr.DataArray, xr.Dataset)):
+        raise TypeError('current must be of type pd.Series, pd.DataFrame, ' + 
+                        f'xr.DataArray, or xr.Dataset. Got{type(current)}')
+
+    # Convert inputs to xr.Dataset
+    voltage = _convert_to_dataset(voltage, 'voltage')
+    current = _convert_to_dataset(current, 'current')
+
+    # Check that sizes are the same
+    if not len(voltage.data_vars) == 3:
         raise ValueError('voltage must have three columns')
-    if not len(current.columns) == 3:
+    if not len(current.data_vars) == 3:
         raise ValueError('current must have three columns')
-    if not current.shape == voltage.shape:
+    if not current.sizes == voltage.sizes:
         raise ValueError('current and voltage must be of the same size')
-    
 
-    abs_current = np.abs(current.values)
-    abs_voltage = np.abs(voltage.values)
+    power = dc_power(voltage, current, to_pandas=False)['Gross']
+    power.name = 'Power'
+    power = power.to_dataset() # force xr.DataArray to be consistently in xr.Dataset format
```

Comment:
This forcing seems like a intermediate step needed prior to getting dc_power to return an xarray.

Is this still needed when we have `to_pandas=False`?

Reply:
When the Gross power variable is extracted from the results of dc_power, the xr.Dataset automatically becomes an xr.DataArray since there's just one variable. I originally added this conversion to a xr.Dataset to prevent that type change.

There shouldn't be any issues with returning a single variable array like pd.Series or xr.DataArray, but the previous function and most of the current ones I'm working with return pd.DataFrame and xr.Dataset